### PR TITLE
mssw Chapter 3 - decorator

### PR DESCRIPTION
**Pull request by Spinodal**

### CODE REVIEW

File: mssw/src/org/mssw/decorator/Generator.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		baseWeapons.add(new Rifle());
+		baseWeapons.add(new Shotgun());
+		baseWeapons.add(new RocketLauncher());
+		baseWeapons.add(new SMG());
+		baseWeapons.add(new PointyStick());
+	}
+
+	public void initialiseCombatents() {
+		combatents = new ArrayList<>();
+		combatents.add(new Combatent("Axp"));
```
Comment (a-panayotopoulos) [inline mssw/src/org/mssw/decorator/Generator.java]: I'd like a double-barrelled pointy stick, please.

File: mssw/src/org/mssw/decorator/Generator.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	public List<Class<? extends Weapon>> createModSet() {
+		List<Class<? extends Weapon>> mods = new ArrayList<>();
+		mods.add(DoubleBarrel.class);
+		mods.add(SniperScope.class);
+		mods.add(Silencer.class);
+		mods.add(Rusty.class);
+
+		return mods;
+	}
```
Comment (Utopia1138) [inline mssw/src/org/mssw/decorator/Generator.java]: With Java 8 I'd be tempted to change this to:
```java
List<UnaryOperator<Weapon>> mods = new ArrayList<>();
mods.addAll(Arrays.asList(
    DoubleBarrel::new, SniperScope::new, Silencer::new, Rusty::new
    ));
return mods;
```

File: mssw/src/org/mssw/decorator/Generator.java (hunk 1 above)
Comment (Spinodal) [inline mssw/src/org/mssw/decorator/Generator.java]: I prefer mine silenced!

**Comment by Utopia1138** [comment]
Ummm...

```
Sdt shot at Spg with his Rocket Launcher
Spg died...
Spg shot at Sdt but missed!
```

**Comment by SimonGross** [comment]
:ghost:

**Comment by Spinodal** [comment]
everyone gets a chance to kill another player from the grave if they were killed before their turn!

**Pull request merged**